server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -267,4 +278,104 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
             }
         });
     }
+
+    void setRoutingWeights(
+        String awarenessAttributeName,
+        Map<String, Double> awarenessAttributeToWeightMap,
+        ActionListener<ClusterPutWeightedRoutingResponse> listener
+    ) {
+        // WRR transport action will validate invalid weights
+        final ClusterPutWeightedRoutingRequest clusterPutRoutingWeightRequest = new ClusterPutWeightedRoutingRequest(
+            awarenessAttributeName
+        );
+        clusterPutRoutingWeightRequest.setWeightedRouting(new WeightedRouting(awarenessAttributeName, awarenessAttributeToWeightMap));
+
+        transportService.sendRequest(
+            transportService.getLocalNode(),
```

Comment:
Should we assert local node is cluster manager node?

Reply:
Done.